source/Missions/Services/Agents/Extensions/AgentExtensions.cs

```diff
@@ -1,13 +1,19 @@
 ﻿using TaleWorlds.MountAndBlade;
 using Missions.Services.Network;
+using System.Linq;
 
 namespace Missions.Services.Agents.Extensions
 {
     public static class AgentExtensions
     {
-        public static bool IsNetworkAgent(this Agent agent)
+        public static bool IsNetworkAgent(this Agent agent) 
         {
             return NetworkAgentRegistry.Instance.AgentToId.ContainsKey(agent);
         }
+
+        public static bool IsPlayerAgent(this Agent agent)
+        {
+            return NetworkAgentRegistry.Instance.PlayerAgents.Any(kvp => kvp.Value == agent);
```

Comment:
O(1) is possible here
```suggestion
    if (NetworkAgentRegistry.Instance.AgentToId.TryGetValue(agent, out Guid id)
    {
            return NetworkAgentRegistry.Instance.PlayerAgents.Contains(id);
    }
    
    return false;
```